app/models/topic.rb

```diff
@@ -99,7 +99,7 @@ def thumbnail_info(enqueue_if_missing: false, extra_sizes: [])
   def generate_thumbnails!(extra_sizes: [])
     return nil unless SiteSetting.create_thumbnails
     return nil unless original = image_upload
-    return nil unless original.filesize < SiteSetting.max_image_size_kb.kilobytes
+    return nil if original.filesize >= SiteSetting.max_image_size_kb.kilobytes
     return nil unless original.width && original.height
```

Comment:
How did this one get through the filter? 🤔

Reply:
See https://github.com/discourse/discourse/pull/20329#discussion_r1108515551